specification/behaviors.md

```diff
@@ -51,3 +51,69 @@ metrics).
 
 See [integration_context.md](integration_context.md) for specifics about
 exchanging additional context between AppD and splunk-otel based agents.
+
+## Trace Snapshot Profiling
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+This section describes the behavior for Splunk instrumentation libraries 
+that contain trace snapshot profiling features.
+
+### Instrumentation Source
+
+Agents MUST specify set the `profiling.instrumentation.source` value to `snapshot`
+
+### Starting Trace Profiler
+
+The OpenTelemetry Baggage entry for `splunk.snapshot.volume` MUST be used to 
+decide whether to profile a trace. A value of `higest` is the signal to begin 
+profiling where as a value of `off` is an explicit signal to not profile.
+
+When profiling a trace, the profiler SHOULD be started when an entry span is 
+detected. An entry span is defined as either the root span of the trace or 
+any other span within a trace whose parent span is remote.
+
+When a trace is profiled agents MUST add the span attribute `splunk.snapshot.profiling` 
+with a value of `true` to the entry span.
+
+Agents SHOULD take an initial stack trace sample when starting to profile a trace.
+
+### Trace Profiling
+An instrumentation library that has trace snapshot profiling capabilities MUST
+be able to sample call stacks for specific trace ids at a fixed interval.
+
+When a language runtime supports threading, stacks MUST be sampled only for 
+trace ids selected for snapshotting. The samples for profiled threads SHOULD be 
+taken instantaneously and MAY be taken at separate times.
```

Comment:
> The samples for profiled threads SHOULD be taken instantaneously and MAY be taken at separate times.

I don't follow the meaning of this sentence. Also taking a stack trace is not an instantaneous operation by any means. In java for example stack trace is taken at a safepoint, which means that all threads are suspended (apparently recent vms are able to suspend only a single thread, but I don't know whether it is used when taking a stack trace).